taskvine/src/manager/vine_manager.c

```diff
@@ -4233,6 +4242,8 @@ static struct vine_task *vine_wait_internal(struct vine_manager *q, int timeout,
 				if(temp->state==VINE_TASK_RETRIEVED) {
 					t = temp;
 				}
+			} else if((t = list_peek_head(q->retrieved_list))) {
```

Comment:
This doesn't seem right, because this case removes a retrieved task from the list, while the very next case also finds a retrieved task, but **doesn't** remove it from the list.